src/components/Heroes/RankColumns.ts

```diff
@@ -0,0 +1,271 @@
+import heroes from 'dotaconstants/build/heroes.json';
+import constants from 'src/components/constants'
+import { abbreviateNumber, displayHeroId } from '../../utility';
+import { GlobalStrings } from 'src/types/common/GlobalStrings'
+import { Heroes } from 'src/types/Hero/Heroes'
+
+type Props = {
+  tabType: HeroesTabType
+  strings: GlobalStrings
+}
+
+enum HeroesTabType {
+  PRO = 'pro',
+  PUBLIC = 'public'
+}
+
+type ProColumn = {
+  displayName: string,
+  field: string,
+  sortFn: boolean,
+  percentBarsWithValue: (row: Row) => string
+}
+
+type PublicColumn = {
+  displayName: string,
+  field: string,
+  sortFn: boolean,
+  percentBarsWithValue: (row: Row) => string
+  displayIcon: string
+  colColor: string
+}
+
+type HeroColumn = {
+  displayName: string,
+  tooltip: string,
+  field: string
+  displayFn: () => JSX.Element,
+  sortFn: (row: Row) => string
+}
+
+type PreparedColumn = (ProColumn | PublicColumn | HeroColumn) & ColumnAdditions
+
+type ColumnAdditions = {
+  paddingLeft: number,
+  paddingRight: number,
+  tooltip: string
+}
+
+type Row = {
+  [key in string]: number
+}
+
+type ColumnTypes = ProColumn | PublicColumn | HeroColumn
```

Comment:
You usually do not need to add `Type` suffix to type (and types are also usually in singular)

```suggestion
type Column = ProColumn | PublicColumn | HeroColumn
```

Reply:
Agreed on the singular, was just the particular use-case of the function where the type was being used made me consider doing it this way. You're right though, looks cleaner like that